packages/gatsby-plugin-preload-fonts/src/prepare/fetch-routes.js

```diff
@@ -37,6 +37,10 @@ ${red(`err`)} could not establish a connection with the dev server
 
   const routesHash = createContentDigest(routes)
   if (cache.hash === routesHash) {
+    if (isCI()) {
+      process.exit(0)
+    }
+
```

Comment:
Can you please explain what's going on here?

Either I'm missing something or the _only_ real change in this PR is a hard exit if running in CI mode. I'm very confused why a hard zero exit of the whole node process, without any kind of logging / warning / error is the expected behaviour here.

Is this plugin not run in the main `gatsby build` or `gatsby develop` process? Perhaps before it and this is supposed to short-circuit it for CI modes? Maybe add a comment describing why would help?

Reply:
Yes, the whole idea of this PR is to enable running this script before building gatsby site in CI. Otherwise it would would await user input what would hang the process. 

We basically exit the script if it determines that it's not needed to run (no new routes are added). Here's superhawks explanation of the method https://github.com/gatsbyjs/gatsby/issues/20185#issuecomment-574428006

If you feel like this should be commented in the code please tell me what you'd expect from reading it.

Reply:
Something like "This runs before a build/develop and in CI we don't need to so we hard exit early". Try to clarify to the reader (or to somebody arbitrarily looking for hard exits in the code) why this one is benign.

Reply:
thank you @wardpeet for adding comments <3